src/libvirtApi/helpers.ts

```diff
@@ -109,40 +115,98 @@ export const Enum = {
     VIR_MIGRATE_OFFLINE: 1024,
 };
 
+/* Utilities for DBus variants.
+ */
+
+export function get_variant_string(val: Variant): string {
+    if (val.t != "s")
+        throw new Error(`Must be a string but has signature ${val.t}`);
+    return val.v as string;
+}
+
+export function get_variant_number(val: Variant): number {
+    if (!"ynqiuxtd".includes(val.t))
```

Comment:
What is this? With no context I have no clue what this means or where to look it up

Reply:
These are the DBus type signatures for numbers: https://dbus.freedesktop.org/doc/dbus-specification.html#type-system

They show up in `Variant.t`, and the corresponding value in `Variant.v`.